nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
Maybe I'm not describing the approach properly. The suggestion I made does not require an `equals/hashCode` implementation. The idea is to basically re-use the same instance across calls to `isRequired`. Since the `validators` set is per field, the one instance of `RequiredFieldValidator` can be maintained within `LabeledItemController`.

By "seeing a duplicate error message", what I mean is that if the user calls `isRequired` and also manually adds their own `RequiredFieldValidator` instance to the set of validators, what will happen is that there will be duplicate validation errors if the field is empty, which isn't a big problem, since that use case will be very rare in my opinion.